obs/project.go

```diff
@@ -40,13 +40,98 @@ type Project struct {
 	UseForBuild  *UseForBuild `json:"useForBuild,omitempty" xml:"useforbuild,omitempty"`
 }
 
-type Client struct {
-	Client   http.Client
+// OBS is the wrapper struct
+type OBS struct {
+	client  Client
+	options *Options
+}
+
+type obsClient struct {
+	*http.Client
+	*Options
+}
+
+// InvokeOBSEndpoint is the Interface method implementation over obsClient
+func (c *obsClient) InvokeOBSEndpoint(ctx context.Context, method string, url string, xmlData *bytes.Buffer) (*http.Response, error) {
+	//TODO implement me
+	req, err := http.NewRequestWithContext(ctx, method, url, xmlData)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: creating request: %v", err),
```

Comment:
This is used for different calls, so `creating obs project` is not accurate any longer
```suggestion
			Message:        fmt.Sprintf("creating request: %v", err),
```